samanthas_telegram_bot/api_queries/send_get.py

```diff
@@ -0,0 +1,293 @@
+"""Functions for sending data to backend to create entities and get required data in return."""
+import json
+import logging
+
+import httpx
+from telegram import Bot, Update
+from telegram.constants import ParseMode
+
+from samanthas_telegram_bot.conversation.auxil.log_and_report import log_and_report
+from samanthas_telegram_bot.conversation.auxil.send_to_admin_group import send_to_admin_group
+from samanthas_telegram_bot.data_structures.constants import API_URL_PREFIX
+from samanthas_telegram_bot.data_structures.context_types import ChatData, UserData
+
+logger = logging.getLogger(__name__)
+
+STATUS_AT_CREATION_STUDENT_TEACHER = "awaiting_offer"
+STATUS_AT_CREATION_TEACHER_UNDER_18 = "active"
+
+
+# TODO how to best test these functions?  We can't test whether all data was collected by the bot,
+#  but it may make sense to test whether all data is passed to backend (to avoid cases when we
+#  add a new field to model and forget to reflect it here).  Is there some sort of "dry run" mode
+#  in the backend so we can send a post request without the record being created?  Ofc, we can
+#  just create a record and then delete it.
+async def _send_personal_info_get_id(user_data: UserData, bot: Bot) -> int:
+    """Creates a personal info item. This function is meant to be called from other functions."""
+    async with httpx.AsyncClient() as client:
+        r = await client.post(
+            f"{API_URL_PREFIX}/personal_info/",
+            data={
+                "communication_language_mode": user_data.communication_language_in_class,
+                "first_name": user_data.first_name,
+                "last_name": user_data.last_name,
+                "telegram_username": user_data.tg_username if user_data.tg_username else "",
```

Comment:
user_data.tg_username or ""

Reply:
Still can't get used to these things :) They like them in JS too.
Fixed